File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        pre = [1]
        temp = 0
        for n in nums:
            temp += n%2
            if len(pre) > temp:
                pre[-1] += 1
            else:
                pre.append(1)
        ans = 0
        loop = len(pre) - k
        for i in range(loop):
            ans += pre[i]*pre[k+i]
        return ans

File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.cpp
class Solution {
public:
    int numberOfSubarrays(vector<int>& nums, int k) {
        int l = 0, count = 0, ans = 0, odd_count = 0;
        for (auto n: nums){
            if(n%2){
                ++odd_count;
                count = 0;
            }
            while(odd_count == k){
                if(nums[l]%2){
                    --odd_count;
                }
                ++count;
                ++l;
            }
            ans += count;
        }
        return ans;
    }
};